completion/tests/test_models.py

```diff
@@ -142,3 +142,69 @@ def test_submit_batch_completion_without_waffle(self):
         with self.assertRaises(RuntimeError):
             blocks = [(self.block_key, 1.0)]
             models.BlockCompletion.objects.submit_batch_completion(self.user, self.course_key, blocks)
+
+
+class BatchCompletionMethodTests(CompletionSetUpMixin, TestCase):
+    """
+    Test model methods:
+        latest completion per course, and all completions per course
+    """
+    COMPLETION_SWITCH_ENABLED = True
+
+    def setUp(self):
+        super(BatchCompletionMethodTests, self).setUp()
+        self.set_up_completion()
+        self.user = UserFactory.create()
+        self.other_user = UserFactory.create()
+        self.course_key = CourseKey.from_string("edX/MOOC101/2049_T2")
+        self.other_course_key = CourseKey.from_string("course-v1:ReedX+Hum110+1904")
+        self.block_keys = [
+            UsageKey.from_string("i4x://edX/MOOC101/video/{}".format(number)) for number in range(5)
+            ]
+        self.submit_faux_completions()
+
+    def submit_faux_completions(self):
+        """
+        Submit completions for given runtime, run at setup
+        """
+        for idx, block_key in enumerate(self.block_keys[0:3]):
+            models.BlockCompletion.objects.submit_completion(
+                user=self.user,
+                course_key=self.course_key,
+                block_key=block_key,
+                completion=1.0 - (0.2 * idx),
+            )
+
+        for idx, block_key in enumerate(self.block_keys[2:]):  # Wrong user
+
+            models.BlockCompletion.objects.submit_completion(
+                user=self.other_user,
+                course_key=self.course_key,
+                block_key=block_key,
+                completion=0.9 - (0.2 * idx),
+            )
+
+        models.BlockCompletion.objects.submit_completion(  # Wrong course
+            user=self.user,
+            course_key=self.other_course_key,
+            block_key=self.block_keys[4],
+            completion=0.75,
+        )
+
+    def test_get_course_completions(self):
+        # Get all completions for course
+        self.assertEqual(
+            models.BlockCompletion.get_course_completions(self.user, self.course_key),
+            {
+                self.block_keys[0]: 1.0,
+                self.block_keys[1]: 0.8,
+                self.block_keys[2]: 0.6,
+            },
+        )
+
+    def test_get_latest_block_completed(self):
```

Comment:
Can we have another test that `get_latest_block_completed()` returns None if there are no completions for the user in the course?